specs/core/1_new_shards.md

```diff
@@ -0,0 +1,440 @@
+            increase_balance(state, committee[i], get_base_reward(state, committee[i]))
+    
+    assert bls_verify(
+        pubkey=bls_aggregate_pubkeys(signer_keys),
+        message_hash=get_block_root_at_slot(state, state.slot - 1),
+        signature=block.light_client_signature,
+        domain=DOMAIN_LIGHT_CLIENT
+    )
+```
+
+### Epoch transition
+
+```python
+def phase_1_epoch_transition(state):
+    # Slowly remove validators from the "online" set if they do not show up
+    for index in range(len(state.validators)):
+        if state.online_countdown[index] != 0:
+            state.online_countdown[index] = state.online_countdown[index] - 1
+    
+    # Update light client committees
+    if get_current_epoch(state) % LIGHT_CLIENT_COMMITTEE_PERIOD == 0:
+        state.current_light_committee = state.next_light_committee
+        state.next_light_committee = committee_to_compact_committee(state, get_light_client_committee(state, get_current_epoch(state)))
+```
+
+### Fraud proofs
+
+TODO. The intent is to have a single universal fraud proof type, which contains (i) an on-time attestation on shard `s` signing a set of `data_roots`, (ii) an index `i` of a particular data root to focus on, (iii) the full contents of the i'th data, (iii) a Merkle proof to the `shard_state_roots` in the parent block the attestation is referencing, and which then verifies that one of the two conditions is false:
+
+* `custody_bits[i][j] != generate_custody_bit(subkey, block_contents)` for any `j`
+* `execute_state_transition(shard, slot, attestation.shard_state_roots[i-1], hash_tree_root(parent), get_shard_proposer(state, shard, slot), block_contents) != shard_state_roots[i]` (if `i=0` then instead use `parent.shard_state_roots[s][-1]`)
+
+## Shard state transition function
+
+```python
+def shard_state_transition(shard: Shard, slot: Slot, pre_state: Hash, previous_beacon_root: Hash, proposer_pubkey: BLSPubkey, block_data: BytesN[MAX_SHARD_BLOCK_CHUNKS * SHARD_BLOCK_CHUNK_SIZE]) -> Hash:
+    # Beginning of block data is the previous block hash
+    assert block_data[:32] == pre_state.latest_block_hash
+    assert block_data[32:64] == int_to_bytes8(slot) + b'\x00' * 24
+    # Signature check
+    assert len(block_data) >= 160
+    assert bls_verify(
+        pubkey=proposer_pubkey,
+        message_hash=hash_tree_root(block_data[:-96]),
+        signature=block_data[-96:],
+        domain=DOMAIN_SHARD_PROPOSER
+    )
+    # We will add something more substantive in phase 2
+    length = len(block.data.rstrip(b'\x00'))
+    return ShardState(slot=slot, root=hash(pre_state + hash_tree_root(block_data)), gasprice=update_gasprice(pre_state, length), latest_block_hash = hash(block_data))
+```
+
+## Honest committee member behavior
+
+Suppose you are a committee member on shard `shard` at slot `current_slot`. Let `state` be the head beacon state you are building on. Three seconds into slot `slot`, run the following procedure:
```

Comment:
Dividing the slot into 4

- beacon block proposal
- shard block proposal
- attestation broadcast
- aggregate broadcast

```suggestion
Suppose you are a committee member on shard `shard` at slot `current_slot`. Let `state` be the head beacon state you are building on. `2 * SECONDS_PER_SLOT // 4` seconds into slot `slot`, run the following procedure:
```